go/vt/tabletserver/query_executor.go

```diff
@@ -106,8 +106,46 @@ func (qre *QueryExecutor) Execute() (reply *mproto.QueryResult) {
 			conn := qre.getConn(qre.qe.connPool)
 			defer conn.Recycle()
 			reply = qre.execSQL(conn, qre.query, true)
+		}
+		if reply != nil {
+			return reply
+		}
+		// autocommit for single dml
+		transactionID := qre.qe.txPool.Begin(qre.ctx)
+		qre.logStats.AddRewrittenSql("begin", time.Now())
+		isQueryFailed := false
+		defer func() {
+			if !isQueryFailed {
+				qre.qe.Commit(qre.ctx, qre.logStats, transactionID)
+				qre.logStats.AddRewrittenSql("commit", time.Now())
+			}
```

Comment:
I think we need to rollback on failure here. That is, if any of the exec functions fail.